File: 자연수뒤집어배열로만들기.py
# 자연수 n을 뒤집어 각 자리 숫자를 원소로 가지는 배열 형태로 리턴해주세요. 
# 예를들어 n이 12345이면 [5,4,3,2,1]을 리턴합니다.

# 접근 방법
# 1. int -> str 변환
# 2. 변환된 n을 reversed시킴
# 3. reversed 시킨 n을 for문을 돌려서 각각 array에 넣어줌 
# 4. array에 넣어주기 전에 str을 다시 int로 변환시켜줌 
# 5. return array 시켜주기 

# 연습 풀이
n = 12345
array = []
a = reversed(str(n))
# print(a)

for i in a : 
    array.append(int(i)) # int로 변환하는 과정에서 막힘 
    print(array)
    

# 최종 풀이 
n = 12345
array = []
def solution(n) : 
    for i in reversed(str(n)):
        array.append(int(i)) # int로 변환하는 과정에서 막힘 
    return array

File: sparta/day5/자연수 뒤집어 배열로 만들기.js
/**
 1. 문제 설명 
 자연수 n을 뒤집어 각 자리 숫자를 원소로 가지는 배열 형태로 리턴해주세요. 
 예를들어 n이 12345이면 [5,4,3,2,1]을 리턴합니다.

 2. 문제 접근 
 - 숫자 형태를 문자열로 변경시켜줌
 - 각 원소들을 array 형태로 담아줌 
 - 각 원소가 담긴 배열을 내림차순 정렬해줌 

 **/

 function solution(n){

    // 배열로 만들기 위해서 숫자 -> 문자 형태로 형변환 
     let str = String(n)

     // 각 원소들을 arr에 담아줌 
     let arr = Array.from(str)

     // 원소들을 reverse()를 이용하여 뒤집고 각 원소들을 number로 변환한 새로운 형태의 배열을 생성 
     let result = arr.reverse().map(Number);
    return result;
 }

 let n = 12345
 console.log(solution(n))